hazelcast-client-legacy/src/test/java/com/hazelcast/client/cache/nearcache/ClientCacheSerializationCountTest.java

```diff
@@ -0,0 +1,266 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.cache.nearcache;
+
+
+import com.hazelcast.cache.ICache;
+import com.hazelcast.cache.impl.nearcache.NearCache;
+import com.hazelcast.cache.impl.nearcache.NearCacheManager;
+import com.hazelcast.client.cache.impl.HazelcastClientCacheManager;
+import com.hazelcast.client.cache.impl.HazelcastClientCachingProvider;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.HazelcastClientProxy;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.NearCacheConfig;
+import com.hazelcast.config.SerializationConfig;
+import com.hazelcast.nio.serialization.ClassDefinition;
+import com.hazelcast.nio.serialization.ClassDefinitionBuilder;
+import com.hazelcast.nio.serialization.Portable;
+import com.hazelcast.nio.serialization.PortableFactory;
+import com.hazelcast.nio.serialization.PortableReader;
+import com.hazelcast.nio.serialization.PortableWriter;
+import com.hazelcast.test.AssertTask;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.cache.spi.CachingProvider;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class ClientCacheSerializationCountTest extends HazelcastTestSupport {
+
+    static final String CACHE_NAME = randomString();
+    TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+    NearCache nearCache;
+    ICache<String, SerializationCountingData> cache;
+
+    private static final AtomicInteger serializeCount = new AtomicInteger();
+    private static final AtomicInteger deserializeCount = new AtomicInteger();
+
+    static class SerializationCountingData implements Portable {
+        public static int FACTORY_ID = 1;
+        public static int CLASS_ID = 1;
+
+        public SerializationCountingData() {
+
+        }
+
+        @Override
+        public int getFactoryId() {
+            return FACTORY_ID;
+        }
+
+        @Override
+        public int getClassId() {
+            return CLASS_ID;
+        }
+
+        @Override
+        public void writePortable(PortableWriter writer) throws IOException {
+            serializeCount.incrementAndGet();
+        }
+
+        @Override
+        public void readPortable(PortableReader reader) throws IOException {
+            deserializeCount.incrementAndGet();
+        }
+
+    }
+
+    @After
+    public void tearDown() {
```

Comment:
`serializeCount` and `deserializeCount` should be reset also here since in case of error, they might left unresetted and this may lead to fail of other tests.